File: Patrones Comportamiento/State/Ejemplo.py
class Strategy:
    def execute(self, a, b): pass

class Add(Strategy):
    def execute(self, a, b):
        return a + b

class Context:
    def __init__(self, strategy):
        self.strategy = strategy
    def run(self, a, b):
        return self.strategy.execute(a, b)

print(Context(Add()).run(2, 3))

File: Patrones Comportamiento/Memento/Ejemplo.js
class Strategy {
  execute(a, b) {}
}

class Add extends Strategy {
  execute(a, b) { return a + b; }
}

class Context {
  constructor(strategy) { this.strategy = strategy; }
  run(a, b) { return this.strategy.execute(a, b); }
}

console.log(new Context(new Add()).run(2, 3));
